### PR TITLE
Add process namespaces feature #3471

### PR DESCRIPTION
**Pull request by bvsubhash**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #3471
| License       | MIT

### CODE REVIEW

File: lib/API/CliUx.js [hunk 1]
```diff
@@ -90,6 +91,7 @@ UX.describeTable = function(proc) {
   safe_push(table,
             { 'status' : colorStatus(pm2_env.status) },
             { 'name': pm2_env.name },
+            { 'namesapce': pm2_env.namespace },
```
Comment (Unitech) [inline lib/API/CliUx.js]: typo namesapce

**Comment by Unitech** [comment]
That's cool!

File: lib/API/CliUx.js (hunk 1 above)
Comment (bvsubhash) [reply lib/API/CliUx.js]: Fixed it, thanks.

**Comment by Unitech** [comment]
That's sick!!
I've added an option to PM2 CLI to specify in which namespace you can start the application so you can do it both via JSON or via CLI:

```bash
pm2 start app.js --name app1 --namespace frontal
pm2 start app.js --name app2 --namespace frontal
pm2 start worker.js --name app3 --namespace workers
## Restart only applications in "frontal" namespace 
pm2 restart frontal
```

**Pull request merged**